.github/workflows/styler_lintr.yml

```diff
@@ -0,0 +1,62 @@
+name: Improving Readability
+on:
+  pull_request:
+    branches: 
+      - "develop"
+      - "main"
+
+jobs:
+  styling-linting-the-code:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0  # Fetch all history to get the previous commit
+
+    - name: Set up R library environment using custom actions
+      uses: r-lib/actions/setup-r@v2
+
+    - name: Install Lintr, Styler and Roxygen2
+      run: |
+        Rscript -e "install.packages(c('lintr', 'styler', 'roxygen2'), repos = 'https://cran.rstudio.com/')"
+
+    - name: Style R code
+      run: |
+        Rscript -e "styler::style_pkg(include_roxygen_examples = FALSE); warnings()"
+
+    - name: Check for changes by styler
+      id: check_styler_changes
+      run: |
+        if git diff --quiet; then
+          echo "No changes made by styler."
+          echo "changes_made=false" >> $GITHUB_ENV
+        else
+          echo "Changes made by styler."
+          echo "changes_made=true" >> $GITHUB_ENV
+        fi
+
+    - name: Commit and push if styler made a change
+      if: env.changes_made == 'true'
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
@hasruleng I just noticed, how come we have github token here an not for the tests?